asset/asset.go

```diff
@@ -946,13 +946,23 @@ type GroupKey struct {
 	// 	tweakedGroupKey = TapTweak(internalKey, tapTweak)
 	GroupPubKey btcec.PublicKey
 
-	// TapscriptRoot is the root of the Tapscript tree that commits to all
-	// script spend conditions for the group key. Instead of spending an
-	// asset, these scripts are used to define witnesses more complex than
-	// a Schnorr signature for reissuing assets. A group key with an empty
-	// Tapscript root can only authorize reissuance with a signature.
+	// TapscriptRoot represents the root of the Tapscript tree that commits
+	// to all script spend conditions associated with the group key. Instead
+	// of simply authorizing asset spending, these scripts enable more
+	// complex witness mechanisms beyond a Schnorr signature, allowing for
+	// reissuance of assets. A group key with an empty Tapscript root can
+	// only authorize reissuance using a signature.
+	//
+	// In the V1 group key construction, this root is never empty. It always
+	// includes two layers of script leaves that commit to the group
+	// anchor's (genesis) asset ID, ensuring any user-provided Tapscript
+	// root is positioned at level 2.
 	TapscriptRoot []byte
 
+	// CustomTapscriptRoot is an optional tapscript root to graft at the
+	// second level of the tapscript tree, if specified.
+	CustomTapscriptRoot fn.Option[chainhash.Hash]
```

Comment:
Perhaps we should combine these two fields into a new encapsulated struct? This way we hide some of the details, and callers only need to interact with a single field instead of two. 

In particular, for the v1 case, IMO the `TapscriptRoot` field should be private and generated as needed.

Reply:
Yes, that sounds like a great idea to me.

I'm trying to make the minimal changes necessary in this PR, so I would like to make the change you suggested in a new PR. Let me know if you'd like me to work another way.